packages/node_modules/@cerebral/http/src/utils.js

```diff
@@ -33,6 +33,17 @@ export function createResponse(options, resolve, reject) {
           )
         )
         break
+      case 'timeout':
```

Comment:
Would we need to add an extra param to the `HttpProviderError` class?
I think that most of the cases you'd like to treat them as generic, but there can be cases where you want to distinct between `abort`, `error` and `timeout` errors in paths.

I think adding a `type` param at the beginning  would be more efficient than the last arg (`isAborted`). I haven't thought too much on what are the implications, but it might bring a breaking change.

Reply:
Alternatively we could create some error subclasses like `HttpAbort`, `HttpTimeout` or `HttpError`, which extend `HttpProviderError`.
This way we can either catch the generic `HttpProviderError` or one of the specific errors.

Reply:
I would suggest adding a type first, as @codrin-iftimie suggests. The only reason is that we do not know the need for separate error flexibility yet. If we do need it that can be implemented later. A typical way to use the types would be:

```js
[
  equals(props`error.type`),
  {
    'timeout': [],
    'whatever': [],
    otherwise: []
  }
]
```

Reply:
Makes sense. :+1:

Reply:
@codrin-iftimie Oki, add a `type` property to **HttpProviderError**. Would be great if you also document it. Let us know if you need any help :)

Reply:
@christianalfoni I can take care of the documentation, no problem. I just want to make sure I took the right path here, before that :)

Thanks.